zrml/prediction-markets/src/tests.rs

```diff
@@ -2126,6 +2126,170 @@ fn it_resolves_a_disputed_market_to_default_if_dispute_mechanism_failed() {
+                    ),
+                    Some((dispute_bond, vec![CHARLIE])),
+                );
+            }
+
+            // remove_last_dispute_from_market_ids_per_dispute_block works
+            let removable_market_ids = MarketIdsPerDisputeBlock::<Runtime>::get(dispute_block);
+            assert_eq!(removable_market_ids.len(), 0);
+
+            let market_ids = MarketIdsPerDisputeBlock::<Runtime>::get(
+                now + <Runtime as Config>::GlobalDisputePeriod::get(),
+            );
+            assert_eq!(market_ids, vec![market_id]);
+            assert!(GlobalDisputes::is_started(&market_id));
+            System::assert_last_event(Event::GlobalDisputeStarted(market_id).into());
+
+            assert_noop!(
+                PredictionMarkets::start_global_dispute(Origin::signed(CHARLIE), market_id),
+                Error::<Runtime>::GlobalDisputeAlreadyStarted
+            );
+        }
+    });
+}
+
+#[test]
+fn start_global_dispute_fails_on_wrong_mdm() {
+    ExtBuilder::default().build().execute_with(|| {
+        let end = 2;
+        assert_ok!(PredictionMarkets::create_market(
+            Origin::signed(ALICE),
+            BOB,
+            MarketPeriod::Block(0..2),
+            get_deadlines(),
+            gen_metadata(2),
+            MarketCreation::Permissionless,
+            MarketType::Categorical(<Runtime as Config>::MaxDisputes::get() + 1),
+            MarketDisputeMechanism::Authorized(CHARLIE),
+            ScoringRule::CPMM,
+        ));
+        let market_id = MarketCommons::latest_market_id().unwrap();
+
+        let market = MarketCommons::market(&market_id).unwrap();
+        let grace_period = market.deadlines.grace_period;
+        run_to_block(end + grace_period + 1);
+        assert_ok!(PredictionMarkets::report(
+            Origin::signed(BOB),
+            market_id,
+            OutcomeReport::Categorical(0)
+        ));
+        let dispute_at_0 = end + grace_period + 2;
+        run_to_block(dispute_at_0);
+
+        for i in 1..=<Runtime as Config>::MaxDisputes::get() {
+            assert_ok!(PredictionMarkets::dispute(
+                Origin::signed(CHARLIE),
+                market_id,
+                OutcomeReport::Categorical(i.saturated_into())
+            ));
+            run_blocks(1);
+            let market = MarketCommons::market(&market_id).unwrap();
+            assert_eq!(market.status, MarketStatus::Disputed);
+        }
+
+        #[cfg(feature = "with-global-disputes")]
+        assert_noop!(
+            PredictionMarkets::start_global_dispute(Origin::signed(CHARLIE), market_id),
+            Error::<Runtime>::InvalidDisputeMechanism
+        );
+    });
+}
+
+#[test]
+fn start_global_dispute_works_without_feature() {
+    ExtBuilder::default().build().execute_with(|| {
+        let non_market_id = 0;
+
+        #[cfg(not(feature = "with-global-disputes"))]
+        assert_ok!(PredictionMarkets::start_global_dispute(Origin::signed(CHARLIE), non_market_id));
```

Comment:
I thought this `assert_ok!` check would be sufficient, because in the case, that there is no market present, the first check with feature `with-global-disputes` fails below. But without the feature here, it doesn't check for the market (So a noop). I thought without creating a market but use a non-existing market id it would be checked enough. How would you check, that start_global_disputes doesn't do anything?

Reply:
You can do this by mimicking [`assert_noop!`](https://paritytech.github.io/substrate/master/src/frame_support/lib.rs.html#709-718):

```rust
let storage_root_before = storage_root(StateVersion::V1);
assert_ok!(PredictionMarkets::start_global_dispute(Origin::signed(CHARLIE), non_market_id));
assert_eq!(storage_root_before, storage_root(StateVersion::V1));
```

Reply:
I guess now that `start_global_dispute` errors, you can just use `assert_noop`.